supertokens/supertokens-node
# Email verification tests

## Related Issues
- 

## Changes
- Adds emailVerifyTokenRequest() in utils for testing
- Tests added:
  - generate token API:
     - Call the API with valid input, email not verified
     - Call the API with valid input, email verified and test error
     - Call the API with no session and see the output (should be 401)
     - Call the API with an expired access token and see that try refresh token is returned
     - Provide your own email callback and make sure that is called
   - email verify API:
     - POST:
       - Call the API with valid input
       - Call the API with an invalid token and see the error
       - token is not of type string from input
       - provide a handlePostEmailVerification callback and make sure it's called on success verification
     - GET:
       - Call the API with valid input
       - Call the API with no session and see the error
       - Call the API with an expired access token and see that try refresh token is returned

## Migration:
  - none
